File: mask_app_java/src/main/java/junghoon/jetpack/sample/mask_app_java/adapter/StoreAdapter.java
package junghoon.jetpack.sample.mask_app_java.adapter;

import android.graphics.Color;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;
import java.util.List;

import junghoon.jetpack.sample.mask_app_java.R;
import junghoon.jetpack.sample.mask_app_java.model.Store;

public class StoreAdapter extends RecyclerView.Adapter<StoreAdapter.StoreViewHolder> {

    private List<Store> mItems = new ArrayList<>();
    @NonNull
    @Override
    public StoreViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.store_item, parent, false);
        return new StoreViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull StoreViewHolder holder, int position) {
        Store store = mItems.get(position);
        holder.mStoreNameTextView.setText(store.getName());
        holder.mAddressTextView.setText(store.getAddr());

        String remainStat = "충분";
        String count = "100개 이상";
        int color = Color.GREEN;
        switch (store.getRemainStat()) {
            case "plenty" :
                count = "100개 이상";
                remainStat = "충분";
                color = Color.GREEN;
                break;
            case "some" :
                count = "30개 이상";
                remainStat = "여유";
                color = Color.YELLOW;
                break;
            case "few" :
                count = "2개 이상";
                remainStat = "매진 임박";
                color = Color.RED;
                break;
            case "empty" :
                count = "재고 없음";
                remainStat = "매진";
                color = Color.GRAY;
                break;
        }
        holder.mCountTextView.setText(count);
        holder.mRemainTextView.setText(remainStat);
        holder.mRemainTextView.setTextColor(color);
        holder.mCountTextView.setTextColor(color);
    }

    @Override
    public int getItemCount() {
        return mItems.size();
    }

    public void updateItems(List<Store> items) {
        mItems = items;
        notifyDataSetChanged();
    }

    static class StoreViewHolder extends RecyclerView.ViewHolder {
        TextView mStoreNameTextView;
        TextView mAddressTextView;
        TextView mDistanceTextView;
        TextView mRemainTextView;
        TextView mCountTextView;

        public StoreViewHolder(@NonNull View itemView) {
            super(itemView);
            mStoreNameTextView = itemView.findViewById(R.id.name_textview);
            mAddressTextView = itemView.findViewById(R.id.addr_textview);
            mDistanceTextView = itemView.findViewById(R.id.distance_textview);
            mRemainTextView = itemView.findViewById(R.id.remain_textview);
            mCountTextView = itemView.findViewById(R.id.count_textview);
        }
    }
}

File: mask_app_kotlin/src/main/java/junghoon/jetpack/sample/mask_app/adapter/StoreAdapter.kt
package junghoon.jetpack.sample.mask_app.adapter

import android.graphics.Color
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.databinding.BindingAdapter
import androidx.recyclerview.widget.RecyclerView
import androidx.recyclerview.widget.RecyclerView.ViewHolder
import junghoon.jetpack.sample.mask_app.R
import junghoon.jetpack.sample.mask_app.databinding.StoreItemBinding
import junghoon.jetpack.sample.mask_app.model.Store
import java.util.*

class StoreAdapter : RecyclerView.Adapter<StoreViewHolder>() {
    private var mItems: List<Store> =
        ArrayList<Store>()

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): StoreViewHolder {
        val view: View =
            LayoutInflater.from(parent.context).inflate(R.layout.store_item, parent, false)
        return StoreViewHolder(view)
    }

    override fun onBindViewHolder(holder: StoreViewHolder, position: Int) {
        holder.binding.store = mItems[position]

    }

    override fun getItemCount(): Int {
        return mItems.size
    }

    fun updateItems(items: List<Store>) {
        mItems = items
        notifyDataSetChanged()
    }
}


class StoreViewHolder(itemView: View) : ViewHolder(itemView) {
    val binding = StoreItemBinding.bind(itemView)
}

@BindingAdapter("remainStat")
fun setRemainStat(textView: TextView, store: Store) = when (store.remain_stat) {
    "plenty" -> textView.text = "충분"
    "some" -> textView.text = "여유"
    "few" -> textView.text = "매진 임박"
    else -> textView.text = "재고 없음"
}

@BindingAdapter("count")
fun setCount(textView: TextView, store: Store) = when (store.remain_stat) {
    "plenty" -> textView.text = "100개 이상"
    "some" -> textView.text = "30개 이상"
    "few" -> textView.text = "2개 이상"
    else -> textView.text = "재고 없음"
}

@BindingAdapter("color")
fun setColor(textView: TextView, store: Store) = when (store.remain_stat) {
    "plenty" -> textView.setTextColor(Color.GREEN)
    "some" -> textView.setTextColor(Color.YELLOW)
    "few" -> textView.setTextColor(Color.RED)
    else -> textView.setTextColor(Color.GRAY)
}


